project-koku/masu
# Handling missing report in payload error to ensure listener event loop is unblocked

An error has been seen in CI and production where it appears that a report file is missing from a payload.  Root cause on that issue is still TBD but it did identify one error path that wasn't handled in the Masu-listener file download/processing flow.

In order to get more information on the missing report file problem I am now extracting these files to a temporary directory on the OpenShift PVC.  That way if the pod goes down we will still have some forensics to look at to get closer to a root cause.  These files will have to be manually removed after triage.

The issue was reproduced by removing the storage report from a payload.  Now the listener will continue to be able to pull in new report payloads and correctly returns a failure to the upload service.

**Testing**
1.  Generate a payload with both files
2. Generate a payload with storage report missing
3.  Add an OCP provider
4. Upload tarball with the missing storage report and verify that masu-listener rejects the file to the upload service and that the temporary contents are in /var/tmp/masu.
5. Upload the payload with all csv files and verify report processing completes and that the temporary extraction directory does not exist but instead files are in the expected insights_local directory

**Test Results**
[masu_listener_block_ut.txt](https://github.com/project-koku/masu/files/3292126/masu_listener_block_ut.txt)


Addresses #464
